addons/panku_console/modules/variable_tracker/module.gd

```diff
@@ -1,54 +1,128 @@
 class_name PankuModuleVariableTracker extends PankuModule
+## Module to register and update some common environments.
+##
+## On module startup current scene root node registered as 'current' environment var,
+## for more convenient access from interactive shell. Module constantly monitoring node
+## tree afterwards and auto rebind 'current' environment in case of current scene changed.
+## Also all user autoload singletons registered with its root node names.
+
+const PROJECT_AUTOLOAD_PREFIX := "autoload/"
+const CURRENT_SCENE_ENV := "current"
+const SHELL_MODULE_NAME := "interactive_shell"
+const DEFAULT_TRACKING_DELAY := 0.5
+
+const CURRENT_REGISTERED_TIP := "[tip] Node '%s' registered as current scene, you can access it by [b]%s[/b]."
+const CURRENT_REMOVED_TIP := "[tip] No current scene found, [b]%s[/b] keyword is no longer available."
+const USER_AUTOLOADS_TIP := "[tip] Accessible user singleton modules: [b]%s[/b]"
 
-# The current scene root node, which will be updated automatically when the scene changes.
-# current scene node was determined by node tree index at the beginning.
 var _current_scene_root:Node
-var _current_scene_index := 0
+var _user_singleton_files := []
 var _tween_loop:Tween
+var _loop_call_back:CallbackTweener
 
 
 func init_module():
-	_current_scene_index = core.get_tree().root.get_child_count() - 1
-	await core.get_tree().process_frame
-	setup_scene_root_tracker()
-	setup_autoload_tracker()
-	print_to_interactive_shell_window()
-
-# always register the current scene root as `current`
-func setup_scene_root_tracker():
-	_current_scene_root = get_scene_root()
-	core.gd_exprenv.register_env("current", _current_scene_root)
+	get_module_opt().tracking_delay = load_module_data("tracking_delay", DEFAULT_TRACKING_DELAY)
+	await core.get_tree().process_frame # not sure if it is necessary
+
+	_update_project_singleton_files()
+	_setup_scene_root_tracker()
+	_check_autoloads()
+
+
+# Parse project setting and collect and autoload files.
+func _update_project_singleton_files() -> void:
+	_user_singleton_files.clear()
+	for property in ProjectSettings.get_property_list():
+		if property.name.begins_with(PROJECT_AUTOLOAD_PREFIX):
+			_user_singleton_files.append(ProjectSettings.get_setting(property.name).trim_prefix("*"))
+
+
+# Check if given node is autoload singleton.
+func _is_singleton(node: Node) -> bool:
+	# Comparing scene file and script file with list of autoload files
+	# from project settings. I'm not sure that approach hundred percent perfect,
+	# but it works so far.
+	if node.scene_file_path in _user_singleton_files:
+		return true
+
+	var script = node.get_script()
+	if script and (script.get_path() in _user_singleton_files):
+		return true
+
+	return false
+
```

Comment:
So, there's actually no elegant way to get a list of running autoload singletons...

Reply:
Unfortunately, I just can't find anything better so far.